### PR TITLE
Add BIP47 to StorageNym; Add storagebip47channellist verify tests

### PR DESCRIPTION
**Pull request by kmels**

### CODE REVIEW

File: include/opentxs-proto/verify/VerifyStorage.hpp [hunk 1]
```diff
@@ -91,21 +92,32 @@ static const VersionMap StorageNymAllowedHash = {
...
+    {7, {1, 1}}
+};
+
 static const VersionMap StorageNymAllowedBlockchainAccountList{
     {4, {1, 1}},
     {5, {1, 1}},
     {6, {1, 1}},
+    {7, {1, 1}},
 };
 static const VersionMap StorageNymListAllowedHash = {
```
Comment (justusranvier) [inline include/opentxs-proto/verify/VerifyStorage.hpp]: It shouldn't be necessary to change the version of StorageNymList

File: src/verify/storagenym/StorageNym_6.cpp [hunk 2]
```diff
@@ -64,77 +64,8 @@ bool CheckProto_6(const StorageNym& input, const bool silent)
     CHECK_SUBOBJECTS(blockchainaccount, StorageNymAllowedBip44Account)
     OPTIONAL_IDENTIFIER(issuers)
     OPTIONAL_IDENTIFIER(paymentworkflow)
-
+    CHECK_NONE(bip47channelindex)
```
Comment (justusranvier) [inline src/verify/storagenym/StorageNym_6.cpp]: Versions 1 through 5 also need these checks

File: include/opentxs-proto/verify/VerifyStorage.hpp (hunk 1 above)
Comment (kmels) [reply include/opentxs-proto/verify/VerifyStorage.hpp]: Done

File: src/verify/storagenym/StorageNym_6.cpp (hunk 2 above)
Comment (kmels) [reply src/verify/storagenym/StorageNym_6.cpp]: Done

File: tests/storagenym/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+set(name unittests-opentxs-storagenym)
+
+set(cxx-sources
+        main.cpp
+        Test_StorageNym.cpp
+        )
+
+include_directories(
```
Comment (justusranvier) [inline tests/storagenym/CMakeLists.txt]: This statement is unnecessary and causes build errors because not all of these directories exist.

**Pull request merged**